deployment/ccip/changeset/state.go

```diff
@@ -562,32 +584,42 @@ func (s CCIPOnChainState) SupportedChains() map[uint64]struct{} {
 
 func (s CCIPOnChainState) View(chains []uint64) (map[string]view.ChainView, error) {
 	m := make(map[string]view.ChainView)
+	mu := sync.Mutex{}
+	grp := errgroup.Group{}
 	for _, chainSelector := range chains {
-		chainInfo, err := deployment.ChainInfo(chainSelector)
-		if err != nil {
-			return m, err
-		}
-		if _, ok := s.Chains[chainSelector]; !ok {
-			return m, fmt.Errorf("chain not supported %d", chainSelector)
-		}
-		chainState := s.Chains[chainSelector]
-		chainView, err := chainState.GenerateView()
-		if err != nil {
-			return m, err
-		}
-		name := chainInfo.ChainName
-		if chainInfo.ChainName == "" {
-			name = strconv.FormatUint(chainSelector, 10)
-		}
-		chainView.ChainSelector = chainSelector
-		id, err := chain_selectors.GetChainIDFromSelector(chainSelector)
-		if err != nil {
-			return nil, fmt.Errorf("failed to get chain id from selector %d: %w", chainSelector, err)
-		}
-		chainView.ChainID = id
-		m[name] = chainView
+		var name string
+		var chainView view.ChainView
+		chainSelector := chainSelector
+		grp.Go(func() error {
+			chainInfo, err := deployment.ChainInfo(chainSelector)
+			if err != nil {
+				return err
+			}
+			if _, ok := s.Chains[chainSelector]; !ok {
+				return fmt.Errorf("chain not supported %d", chainSelector)
+			}
+			chainState := s.Chains[chainSelector]
+			chainView, err = chainState.GenerateView()
+			if err != nil {
+				return err
+			}
+			name = chainInfo.ChainName
+			if chainInfo.ChainName == "" {
+				name = strconv.FormatUint(chainSelector, 10)
+			}
+			chainView.ChainSelector = chainSelector
+			id, err := chain_selectors.GetChainIDFromSelector(chainSelector)
+			if err != nil {
+				return fmt.Errorf("failed to get chain id from selector %d: %w", chainSelector, err)
+			}
+			chainView.ChainID = id
+			mu.Lock()
+			m[name] = chainView
+			mu.Unlock()
+			return nil
+		})
```

Comment:
nit
```suggestion
		grp.Go(func(chainSelector uint64) error {
			chainInfo, err := deployment.ChainInfo(chainSelector)
			if err != nil {
				return err
			}
			if _, ok := s.Chains[chainSelector]; !ok {
				return fmt.Errorf("chain not supported %d", chainSelector)
			}
			chainState := s.Chains[chainSelector]
			chainView, err = chainState.GenerateView()
			if err != nil {
				return err
			}
			name = chainInfo.ChainName
			if chainInfo.ChainName == "" {
				name = strconv.FormatUint(chainSelector, 10)
			}
			chainView.ChainSelector = chainSelector
			id, err := chain_selectors.GetChainIDFromSelector(chainSelector)
			if err != nil {
				return fmt.Errorf("failed to get chain id from selector %d: %w", chainSelector, err)
			}
			chainView.ChainID = id
			mu.Lock()
			m[name] = chainView
			mu.Unlock()
			return nil
		}(chainSelector))
```

Can do similar thing for token pool parallel run as well.

Reply:
could you  please elaborate?

Reply:
I referred that for better readability. It may create some confusion when variable is reassigned to another variable with the same name or it needs another glance to understand why we do that!

Per our discussion, even my suggestion might still cause race condition and having local variable might be better.